### PR TITLE
Disable building Proton by default

### LINKED ISSUE #2361: Swap proton build feature from opt-out (--disable-proton) to opt-in (--enable-proton)
Because our proton support is a new feature and likely to be used by a subset of users, (and we've also encountered some build issues for users) let's make it opt-in for the time being.

### PR DESCRIPTION
**Pull request by daira**
fixes #2361

Signed-off-by: Daira Hopwood <daira@jacaranda.org>

### CODE REVIEW

File: zcutil/build.sh [hunk 1]
```diff
@@ -88,11 +88,11 @@ then
...
 
-# If --disable-proton is the next argument, disable building Proton code:
-PROTON_ARG=''
-if [ "x${1:-}" = 'x--disable-proton' ]
+# If --enable-proton is the next argument, enable building Proton code:
+PROTON_ARG='--enable-proton=no'
+if [ "x${1:-}" = 'x--enable-proton' ]
 then
-    PROTON_ARG='--enable-proton=no'
+    PROTON_ARG=''
```
Comment (nathan-at-least) [inline zcutil/build.sh]: Nitpick: I suspect many users do `./configure …; make; make install`, and I'd like to change the default for them. However, I'm content to rely on the underspecified policy that "`build.sh` is the only officially supported build path for Zcash" (despite my desire for #58 eventually).

**Comment by bitcartel** [comment]
Note that #2280 should be included in 1.0.9 because it fixes potential build problems for people who do want to build proton. Otherwise they will opt in to build and be confused as to why it doesn't.

**Comment by nathan-at-least** [comment]
@zkbot r+

**Comment by nathan-at-least** [comment]
Tests failed because I manually cancelled the build, due to a problem with #2174.

**Comment by nathan-at-least** [comment]
Eh, just clicked 'rebuild' on buildbot. Let's see how homu likes that.

**Comment by nathan-at-least** [comment]
The answer is Homu does not like that, so:

@zkbot retry

**Comment by nathan-at-least** [comment]
@zkbot retry

(I've confused buildbot by stopping and restarting various builders without regard to their triggering dependencies.)

**Comment by nathan-at-least** [comment]
@zkbot retry

**Comment by nathan-at-least** [comment]
@zkbot retry

**Comment by str4d** [comment]
Test failed because the builder couldn't access the network parameters.

@zkbot retry

**Comment by str4d** [comment]
The *very last test* failed because the builder didn't have pyzmq installed :joy: 

@zkbot retry

**Pull request merged**